### PR TITLE
#3266 Error in calculating quantity ordered or reserved for order manufacturing

### PR DESCRIPTION
**Pull request by e-Evolution**
…ring order and distribution order https://github.com/adempiere/adempiere/issues/3266

(cherry picked from commit 2d73c48a84409357988b9ba3165945fb5acda837)

### CODE REVIEW

File: base/src/org/eevolution/model/MPPOrder.java [hunk 1]
```diff
@@ -845,11 +839,11 @@ public boolean closeIt()
...
 		for(MPPOrderBOMLine line : getLines())
 		{
-			BigDecimal old = line.getQtyRequired();
-			if (old.compareTo(line.getQtyDelivered()) != 0)
+			BigDecimal qtyRequired = line.getQtyRequired();
+			if (qtyRequired.compareTo(line.getQtyDelivered()) != 0)
 			{	
 				line.setQtyRequired(line.getQtyDelivered());
-				line.addDescription(Msg.parseTranslation(getCtx(), "@closed@ @QtyRequired@ (" + old + ")"));
+				line.addDescription(Msg.parseTranslation(getCtx(), "@closed@ @QtyRequired@ (" + qtyRequired + ")"));
```
Comment (yamelsenih) [inline base/src/org/eevolution/model/MPPOrder.java:846]: Please add a better definition for this, toy can use 
```Java
line.addDescription(Msg.parseTranslation(getCtx(), "@closed@ @QtyRequired@ (" + DisplayType.getNumberFormat(DisplayType.Quantity).format(qtyRequired) + ")"));
```

**Comment by yamelsenih** [comment]
Hi @e-Evolution I can see many things that should be change for accept it

File: base/src/org/eevolution/model/MPPOrder.java (hunk 1 above)
Comment (e-Evolution) [reply base/src/org/eevolution/model/MPPOrder.java:846]: @yamelsenih Hi Yamel!

This code is not new, it is a legacy code, it was only refactored and separated into methods to make the code more readable.

I can modify the legacy code, but this makes it more difficult to follow the changes, I will modify what is possible thus avoiding introducing any problem.

File: base/src/org/compiere/model/MOrder.java [hunk 2]
```diff
@@ -2283,7 +2298,7 @@ public String reopenIt() {
 			}
 		}
 		//	Clear Reservations
-		if (!reserveStock(null, lines))
+		if (!reserveStock(lines))
 		{
 			m_processMsg = "Cannot unreserve Stock (close)";
```
[the remark this answers is not in the document]
Comment (e-Evolution) [reply base/src/org/compiere/model/MOrder.java]: This is legacy code , but I change the message to @Error@ @Qty@ @QtyNotReserved@ that is replace for "Error Quantity Not Reserved"

File: base/src/org/compiere/model/MOrder.java [hunk 3]
```diff
@@ -1308,7 +1308,7 @@ public String prepareIt()
 		//	Lines
 		if (explodeBOM())
 			lines = getLines(true, MOrderLine.COLUMNNAME_M_Product_ID);
-		if (!reserveStock(documentType, lines))
+		if (!reserveStock(lines))
 		{
 			m_processMsg = "Cannot reserve Stock";
```
[the remark this answers is not in the document]
Comment (e-Evolution) [reply base/src/org/compiere/model/MOrder.java]: This is legacy code , but I change the message to @Error@ @Qty@ @QtyNotReserved@ that is replace for "Error Quantity Not Reserved"

File: base/src/org/eevolution/model/MPPOrder.java (hunk 1 above)
Comment (e-Evolution) [reply base/src/org/eevolution/model/MPPOrder.java]: I modify , but the mendable not is very different that https://github.com/adempiere/adempiere/blob/develop/base/src/org/compiere/model/MOrder.java#L2071

**Pull request merged**